internal/gatewayapi/clustersettings.go

```diff
@@ -76,7 +77,7 @@ func translateTrafficFeatures(policy *egv1a1.ClusterSettings) (*ir.TrafficFeatur
 	// If nothing was set in any of the above calls, return nil instead of an empty
 	// container
 	var empty ir.TrafficFeatures
-	if empty == *ret {
+	if reflect.DeepEqual(empty, *ret) {
```

Comment:
why did this change ?

Reply:
invalid operation: empty == *ret (struct containing []*ir.Compression cannot be compared)

Reply:
That's the issue, https://github.com/envoyproxy/gateway/pull/5001#discussion_r1909717242 should solve this

Reply:
My bad, this looks good, sorry for the false alarm